packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -730,26 +733,29 @@ class _RailDestination extends StatelessWidget {
     return Semantics(
       container: true,
       selected: selected,
-      child: Stack(
-        children: <Widget>[
-          Material(
-            type: MaterialType.transparency,
-            clipBehavior: Clip.none,
-            child: InkResponse(
-              onTap: onTap,
-              onHover: (_) {},
-              highlightShape: BoxShape.rectangle,
-              borderRadius: BorderRadius.all(Radius.circular(minWidth / 2.0)),
-              containedInkWell: true,
-              splashColor: colors.primary.withOpacity(0.12),
-              hoverColor: colors.primary.withOpacity(0.04),
-              child: content,
+      child: Padding(
```

Comment:
Also, did we discuss whether we want the padding to be surrounding the tappable InkResponse? I'm not sure what is the most useful behavior here, but we should verify whether to have Padding wrap InkResponse or vice versa.

Reply:
Another optimization that can be made here is that when padding is `EdgeInsets.zero` or `null`, to completely omit including the Padding widget in the widget tree altogether. Something similar to: 
```dart
if (padding != null || padding != EdgeInsets.zero) {
  child = Padding(padding: padding, child: child);
} else {
  // What is previously there
}
```

Reply:
> Also, did we discuss whether we want the padding to be surrounding the tappable InkResponse? I'm not sure what is the most useful behavior here, but we should verify whether to have Padding wrap InkResponse or vice versa.

@shihaohong surrounding with tappable InkResponse would work, although I was thinking of a use case where the user might just want spacing between two items (like 100 or so), would it be okay to have such a long tappable area?



> Another optimization that can be made here is that when padding is `EdgeInsets.zero` or `null`, to completely omit including the Padding widget in the widget tree altogether. Something similar to:
> 
> ```dart
> if (padding != null || padding != EdgeInsets.zero) {
>   child = Padding(padding: padding, child: child);
> } else {
>   // What's there before
> }
> ```

Sure, I'll make the changes :) :+1: 
Also, should I add a test to check that the padding is not added in case of ```EdgeInsets.zero```.

Reply:
Yes, that's a good test to add!